setup/src/main/java/io/jenkins/jenkinsfile/runner/App.java

```diff
@@ -12,12 +14,21 @@
 
     @Override
     public int run(JenkinsLauncherCommand command) throws Throwable {
-        if (!(command instanceof RunJenkinsfileCommand)) {
+        if (!(command instanceof JenkinsfileCommand)) {
             throw new IllegalStateException(
                     String.format("%s is invoked with a wrong class type. Required=%s, got=%s",
-                            App.class, RunJenkinsfileCommand.class, command.getClass()));
+                            App.class, JenkinsfileCommand.class, command.getClass()));
+        }
+
+        JenkinsLauncher<?> launcher;
+        if (command instanceof RunJenkinsfileCommand) {
+            launcher = new JenkinsfileRunnerLauncher((RunJenkinsfileCommand) command);
+        } else if (command instanceof LintJenkinsfileCommand) {
```

Comment:
Later some simplification will be needed for the Command => Launcher mapping. Fine for now

Reply:
Agreed.  I did take a quick look at this.  Thought it would be good if each `JenkinsLauncherCommand` had a factory method that allowed it to create the right type of `JenkinsLauncher`, but that would have introduced a circular dependency between the `bootstrap` and `setup` modules.  Will leave it for now while it's still simple.